Paltineanu Rares-Mihai/Curs7/curs7-3.10/main.c

```diff
@@ -0,0 +1,64 @@
+/**Write a function to calculate the transposed of a matrix,
+using pointers and pointer expressions to access to the elements of the matrix.
+The resulted matrices shall be displayed as a bi-dimensional table.
+*/
+#include <stdio.h>
+#include <stdlib.h>
+
+void readMat(int ***mat, int rows, int col) {
+
+    printf("Read the matrix:\n");
+    for(int i = 0; i < rows; i++) {
+        for(int j = 0; j < col; j++) {
+            scanf("%d", &mat[i][j]);
+        }
+    }
+}
+
+void transposed(int **mat, int ***tr, int rows, int col) {
+
+    for(int i = 0; i < rows; i++) {
+        for(int j = 0; j < col; j++) {
+            tr[j][i] = mat[i][j];
+        }
+    }
+}
+
+void write(int **tr, int rows, int col) {
+
+    for(int i = 0; i < rows; i++) {
+        for(int j = 0; j < col; j++) {
+            printf("%d ", tr[i][j]);
+        }
+        printf("\n");
+    }
+}
+
+int main()
+{
+    int rows, col;
+    printf("Read the rows and the columns: ");
+    scanf("%d %d", &rows, &col);
+
+    int **mat = (int **)malloc(sizeof(int *) * rows);
```

Comment:
As a hint, you could be writing a function to allocate matrices, something like this:
```
int ** allocateMatrix(int rows, int cols) { 
    int ** mat =  (int **)malloc(sizeof(int *) * rows);
    for(int i = 0; i < rows; i++) {
        mat[i] = (int *)malloc(sizeof(int) * col);
    }
    return mat;
}
```
And then end up calling it like:
`int ** mat = allocateMatrix( ... )`

Reply:
Ik, next time i will use it!